compiler-core/src/error.rs

```diff
@@ -58,9 +60,12 @@ pub enum Error {
     Type {
         path: Utf8PathBuf,
         src: EcoString,
-        error: crate::type_::Error,
+        error: Vec1<crate::type_::Error>,
     },
 
+    #[error("type checking module {module} produced no results")]
+    TypeNotGenerated { module: Name },
```

Comment:
I added this type in case we have a compiler bug that causes errors to be empty but no ast to be generated (instead of panicing like before). lmk if that seems reasonable